core/backend/src/IModelDb.ts

```diff
@@ -143,14 +144,17 @@ export abstract class IModelDb extends IModel {
     this.nativeDb.setIModelDb(this);
     this.initializeIModelDb();
     IModelDb._openDbs.set(this._fileKey, this);
-    IModelHost.onBeforeShutdown.addListener(() => {
-      if (this.isOpen) {
-        try {
-          this.abandonChanges();
-          this.close();
-        } catch { }
-      }
-    });
+
+    if (undefined === IModelDb._shutdownListener) { // the first time we create an IModelDb, add a listener to close any orphan files at shutdown.
+      IModelDb._shutdownListener = IModelHost.onBeforeShutdown.addListener(() => {
+        [...IModelDb._openDbs.values()].forEach((db) => {// note: db.close() removes from _openedDbs, so we can't iterate over that

```

Comment:
FWIW you *can* remove items from javascript containers while iterating them.

Reply:
wow, seems crazy. I just presumed you couldn't.

Reply:
fixed